pocs/camera/__init__.py

```diff
@@ -115,71 +115,73 @@ def kwargs_or_config(item, default=None):
 
     primary_camera = None
 
+    # Different models require different connections methods.
+    model_requires = {
+        'canon_gphoto2': 'port',
+        'sbig': 'serial_number',
+        'zwo': 'serial_number',
+        'fli': 'serial_number',
+    }
+
     device_info = camera_info['devices']
     for cam_num, device_config in enumerate(device_info):
         cam_name = 'Cam{:02d}'.format(cam_num)
 
         if not a_simulator:
-            camera_model = device_config.get('model')
-
             # Assign an auto-detected port. If none are left, skip
             if auto_detect:
                 try:
-                    camera_port = ports.pop()
+                    device_config['port'] = ports.pop()
                 except IndexError:
-                    logger.warning(
-                        "No ports left for {}, skipping.".format(cam_name))
+                    logger.warning("No ports left for {}, skipping.".format(cam_name))
                     continue
             else:
+                # Check for proper connection method.
+                model = device_config['model']
                 try:
-                    camera_port = device_config['port']
-                except KeyError:
+                    connection_method = model_requires[model]
+                    if connection_method not in device_config:
+                        raise error.CameraNotFound
+                except (KeyError, error.CameraNotFound):
                     raise error.CameraNotFound(
-                        msg="No port specified and auto_detect=False")
+                        msg=f"No connection_method for {model} specified and auto_detect=False")
 
-            camera_focuser = device_config.get('focuser', None)
-            camera_filterwheel = device_config.get('filterwheel', None)
-            camera_readout = device_config.get('readout_time', 6.0)
+            device_config.setdefault('focuser', None)
+            device_config.setdefault('filterwheel', None)
+            device_config.setdefault('readout_time', 6.0)
```

Comment:
Not actually necessary to set default values for `focuser`, `filterwheel` or `readout_time` here as the `.__init__()` methods of the `Camera` classes will do appropriate things if these are not set. I'd argue against doing it here as it's redundant, and potentially confusing (user may expect the defaults from the `Camera` classes, but instead gets these values).

I'm glad to learn of the existence of the `.setdefault()` dictionary method, though!